server/apps/microservices/src/processors/metadata-extraction.processor.ts

```diff
@@ -338,3 +341,86 @@ export class MetadataExtractionProcessor {
     return Duration.fromObject({ seconds: videoDurationInSecond }).toFormat('hh:mm:ss.SSS');
   }
 }
+
+@Processor(QueueName.SIDECAR)
+export class SidecarProcessor {
+  private logger = new Logger(SidecarProcessor.name);
+  private assetCore: AssetCore;
+
+  constructor(
+    @Inject(IAssetRepository) private assetRepository: IAssetRepository,
+    @Inject(IJobRepository) private jobRepository: IJobRepository,
+  ) {
+    this.assetCore = new AssetCore(assetRepository, jobRepository);
+    this.init();
+  }
+
+  private async init() {
+  }
+
+  @Process(JobName.QUEUE_SIDECAR)
+  async handleQueueSidecar(job: Job<IBaseJob>) {
+    try {
+      const { force } = job.data;
+      const assets = force ? await this.assetRepository.getWith(WithProperty.SIDECAR) : await this.assetRepository.getWithout(WithoutProperty.SIDECAR);
+
+      for (const asset of assets) {
+        const fileName = asset.originalFileName
+        const name = force ? JobName.SIDECAR_SYNC : JobName.SIDECAR_DISCOVERY;
+        await this.jobRepository.queue({ name, data: { asset, fileName } });
+      }
+    } catch (error: any) {
+      this.logger.error(`Unable to queue sidecar scanning`, error?.stack);
+    }
+  }
+
+  @Process(JobName.SIDECAR_DISCOVERY)
+  async handleQueueSidecarDiscovery(job: Job<IAssetUploadedJob>) {
+    let { asset, fileName } = job.data
+    if (!asset.isVisible) {
+      return;
+    }
+
+    try {
+      const sidecarPath = await this.getSidecarPath(asset);
+      if (sidecarPath) {
+        asset = await this.assetCore.save({ id: asset.id, sidecarPath });
+        const name = asset.type === AssetType.VIDEO ? JobName.EXTRACT_VIDEO_METADATA : JobName.EXIF_EXTRACTION;
+        await this.jobRepository.queue({ name, data: { asset, fileName } });
+      }
+    } catch (error: any) {
+      this.logger.error(`Unable to queue metadata extraction`, error?.stack);
+    }
+  }
+
+  @Process(JobName.SIDECAR_SYNC)
+  async handleQueueSidecarSync(job: Job<IAssetUploadedJob>) {
+    const { asset, fileName } = job.data
+    if (!asset.isVisible) {
+      return;
+    }
+
+    try {
+      const name = asset.type === AssetType.VIDEO ? JobName.EXTRACT_VIDEO_METADATA : JobName.EXIF_EXTRACTION;
```

Comment:
This method is very trivial right now because it just queues all assets, but we should re-visit optimizing this to only detected changed files. We can add that in this PR, or I'm fine merging it as is and revisiting that in a subsequent PR. If that's the case, you should add a `// TODO: optimize to only queue assets with recent xmp changes` note.

Reply:
Added a todo here.